### PR TITLE
fix: crash when trying to open a youtube video when offline

### PR DESCRIPTION
**Pull request by dixidroid**
fixed the crash, where the 'encodedVideos' field is not null, but all fields inside are null.

### CODE REVIEW

**Review by volodymyr-chekyrta — DISMISSED** [review]

**Review by k1rill — APPROVED** [review]

File: course/src/main/java/org/openedx/course/presentation/unit/container/CourseUnitContainerAdapter.kt [hunk 1]
```diff
@@ -23,27 +23,17 @@ class CourseUnitContainerAdapter(
 
     private fun unitBlockFragment(block: Block): Fragment {
         return when {
-            (block.isVideoBlock && block.studentViewData?.encodedVideos != null) -> {
+            (block.isVideoBlock &&
+                    (block.studentViewData?.encodedVideos?.videoUrl?.isEmpty() == false ||
+                            block.studentViewData?.encodedVideos?.youtube?.url?.isEmpty() == false)) -> {
```
Comment (HamzaIsrar12) [inline course/src/main/java/org/openedx/course/presentation/unit/container/CourseUnitContainerAdapter.kt:28]: Please create a utility function inside `EncodedVideos` to make it more readable for checking `videoUrl` and `youtube.url`.

File: core/src/main/java/org/openedx/core/domain/model/Block.kt [hunk 2]
```diff
@@ -117,11 +117,11 @@ data class EncodedVideos(
...
-        get() = mobileHigh?.url
-            ?: mobileLow?.url
-            ?: desktopMp4?.url
+        get() = fallback?.url
             ?: hls?.url
-            ?: fallback?.url
+            ?: desktopMp4?.url
+            ?: mobileHigh?.url
+            ?: mobileLow?.url
             ?: ""
```
Comment (HamzaIsrar12) [inline core/src/main/java/org/openedx/core/domain/model/Block.kt:125]: The `videoUrl` for streaming (not downloading) needs to be selected based on the `stream_priority` property of the `VideoInfo` model. Though we can tackle it in a different PR.

**Review by HamzaIsrar12 — CHANGES_REQUESTED** [review]
Minor nits only.

File: course/src/main/java/org/openedx/course/presentation/unit/container/CourseUnitContainerAdapter.kt (hunk 1 above)
Comment (dixidroid) [reply course/src/main/java/org/openedx/course/presentation/unit/container/CourseUnitContainerAdapter.kt]: done

File: core/src/main/java/org/openedx/core/domain/model/Block.kt (hunk 2 above)
Comment (dixidroid) [reply core/src/main/java/org/openedx/core/domain/model/Block.kt:125]: the videoUrl field uses only in 2 places - in CourseUnitContainerAdapter (where I took the condition) and in CourseUnitContainerFragment where we have only one check - !encodedVideo.videoUrl.endsWith(“.m3u8”)

so this field is not used for offline purposes

Comment (HamzaIsrar12) [reply core/src/main/java/org/openedx/core/domain/model/Block.kt:125]: Yeap. Just to clarify, the `stream_priority` field is used to select a video URL from all the other URLs in the encoded video object based on the priority configured on the backend side.

Comment (HamzaIsrar12) [reply core/src/main/java/org/openedx/core/domain/model/Block.kt:125]: We can ignore this for now.

Comment (dixidroid) [reply core/src/main/java/org/openedx/core/domain/model/Block.kt:125]: got it

**Review by HamzaIsrar12 — APPROVED** [review]

**Pull request merged**